src/plugins/minimal_scene/MinimalScene.cc

```diff
@@ -561,6 +578,87 @@ void GzRenderer::BroadcastKeyPress()
   this->dataPtr->keyEvent.SetType(common::KeyEvent::NO_EVENT);
 }
 
+/////////////////////////////////////////////////
+rendering::CameraPtr GzRenderer::Camera()
+{
+  return this->dataPtr->camera;
+}
+
+#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 2) && QT_CONFIG(vulkan)
+/////////////////////////////////////////////////
+/// \brief fillQtInstanceExtensionsToOgre
+/// Extract Vulkan Instance extension information to be sent to OgreNext
+/// \param[in] inst Qt's Vulkan Instance to extract
+/// \param[out] externalInstance Data to be sent to OgreNext
+static void fillQtInstanceExtensionsToOgre(
+  const QVulkanInstance *inst,
+  rendering::GzVulkanExternalInstance &externalInstance)
+{
+  {
+    QByteArrayList extensions = inst->extensions();
+
+    for (const auto &ext : extensions)
+    {
+      externalInstance.instanceExtensions.push_back(VkExtensionProperties{});
+      VkExtensionProperties &extProp =
+        externalInstance.instanceExtensions.back();
+      strncpy(extProp.extensionName, ext.constData(),
+              VK_MAX_EXTENSION_NAME_SIZE);
+      extProp.extensionName[VK_MAX_EXTENSION_NAME_SIZE - 1u] = 0;
+    }
+  }
+
+  {
+    QByteArrayList layers = inst->layers();
+
+    for (const auto &layer : layers)
+    {
+      externalInstance.instanceLayers.push_back(VkLayerProperties{});
+      VkLayerProperties &layerProp = externalInstance.instanceLayers.back();
+      strncpy(layerProp.layerName, layer.constData(),
+              VK_MAX_EXTENSION_NAME_SIZE);
+      layerProp.layerName[VK_MAX_EXTENSION_NAME_SIZE - 1u] = 0;
+    }
+  }
+}
+
+/////////////////////////////////////////////////
+/// \brief fillQtDeviceExtensionsToOgre
+/// Extract Vulkan Device extension info to be sent to OgreNext
+/// This data is obtained from Environment variables
+/// \param[out] externalDevice Data to be sent to OgreNext
+static void fillQtDeviceExtensionsToOgre(
+  rendering::GzVulkanExternalDevice &externalDevice)
+{
+  // We know Qt adds these by looking at
+  //  qt-everywhere-src-5.15.2/qtbase/src/gui/rhi/qrhivulkan.cpp
+  QVector<QString> deviceExtensions;
+  deviceExtensions.append("VK_KHR_swapchain");
+
+  QByteArray envExts = qgetenv("QT_VULKAN_DEVICE_EXTENSIONS");
+  if (!envExts.isEmpty())
+  {
+    QByteArrayList envExtList = envExts.split(';');
+    for (const auto &ext : envExtList)
+    {
+      if (!ext.isEmpty())
+      {
+        deviceExtensions.append(ext);
+      }
+    }
+  }
+
+  for (const auto &ext : deviceExtensions)
+  {
+    externalDevice.deviceExtensions.push_back(VkExtensionProperties{});
+    VkExtensionProperties &extProp = externalDevice.deviceExtensions.back();
+    strncpy(extProp.extensionName, ext.toStdString().c_str(),
+            VK_MAX_EXTENSION_NAME_SIZE);
```

Comment:
there is a compile warning here:

```
In function ‘char* strncpy(char*, const char*, size_t)’,
    inlined from ‘void fillQtDeviceExtensionsToOgre(gz::rendering::v8::GzVulkanExternalDevice&)’ at /home/ian/code/gz_h_ws/src/gz-gui/src/plugins/minimal_scene/MinimalScene.cc:655:12,
    inlined from ‘std::string gz::gui::plugins::GzRenderer::Initialize(gz::gui::plugins::RenderThreadRhi&)’ at /home/ian/code/gz_h_ws/src/gz-gui/src/plugins/minimal_scene/MinimalScene.cc:711:35:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:34: warning: ‘char* __builtin_strncpy(char*, const char*, long unsigned int)’ specified bound 256 equals destination size [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
   96 |                                   __glibc_objsize (__dest));

```sh

Reply:
That should be impossible? The macros:

```cpp
#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 2) && QT_CONFIG(vulkan)
#endif
```

Which means it should be excluded from complication with Qt 5.13.3

Anyway, the macro says we are doing this:

```cpp
char data[256];
strncpy( data, somePtr, 256);
```

Which is dangerous because if strlen( somePtr ) >= 256; then data[255] will not contain the null terminator `\0`.

Which is why I do on the next line:

```cpp
extProp.extensionName[VK_MAX_EXTENSION_NAME_SIZE - 1u] = 0;
```

in order to enforce the string is always valid.

Anyway, there may be other ways to fix this warning:

```cpp
strncpy(extProp.extensionName, ext.constData(),
              VK_MAX_EXTENSION_NAME_SIZE - 1u);
```

Reply:
ah typo, I was testing with Qt 5.1**5**.3 not Qt 5.1**3**.3. I added the suggested fix to that line (and left other similar lines untouched) to get rid of the warning.  99f887f